mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/ExtensionModelFactory.java

```diff
@@ -289,6 +294,15 @@ private void enrichDeclaration(ExtensionLoadingContext extensionLoadingContext)
 
       if (enricher instanceof WalkingDeclarationEnricher) {
         ((WalkingDeclarationEnricher) enricher).getWalkDelegate(extensionLoadingContext).ifPresent(walkDelegates::add);
+      } else if (isAggregatorEnricher(enricher)) {
+        // Enrichers use information of the declaration. Since the routers used to be constructs, to make sure to keep backward
```

Comment:
* extract this to a private method and turn this method into its javadoc
* Add a reference to the GUS ticket
* add a `@since` tag so that it's clear in which version routers went from constructs to operations